hw/ip/otbn/dv/tracer/cpp/log_trace_listener.cc

```diff
@@ -0,0 +1,63 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cassert>
+#include <iomanip>
+#include <ios>
+#include <sstream>
+#include <string>
+
+#include "log_trace_listener.h"
+
+LogTraceListener::LogTraceListener(const std::string &log_filename)
+    : trace_log(log_filename, std::fstream::out) {
+  if (!trace_log.is_open()) {
+    std::ostringstream oss;
+    oss << "Could not open log file: " << log_filename;
+    throw std::runtime_error(oss.str());
+  }
+}
+
+void LogTraceListener::AcceptTraceString(const std::string &trace,
+                                         unsigned int cycle_count) {
+  assert(trace_log.is_open());
+
+  // Split the trace up into a vector of strings, one per line
+  auto trace_lines = SplitTraceLines(trace);
+  bool first_line = true;
+
+  // Write out the lines from the trace
+  for (auto &line : trace_lines) {
+    if (first_line) {
+      // Special handling for the first line
```

Comment:
Nit: I'm not sure this comment adds value (obviously this is special handling for the first line: the comment is at the start of an `if` statement whose test is `first_line`!)